tsdb/db.go

```diff
@@ -1311,17 +1310,45 @@ func (db *DB) Querier(_ context.Context, mint, maxt int64) (storage.Querier, err
 		return nil, errors.Wrapf(err, "open querier for block %s", b)
 	}
 
-	if len(OverlappingBlocks(blockMetas)) > 0 {
-		return &verticalQuerier{
-			querier: querier{
-				blocks: blockQueriers,
-			},
-		}, nil
+	return newQuerier(blockQueriers), nil
+}
+
+// Querier returns a new querier over the data partition for the given time range.
+// A goroutine must not handle more than one open Querier.
```

Comment:
I don't think this is what you meant to say

Reply:
Copied comment, sorry. FIxing.

Reply:
If you mean

 > // A goroutine must not handle more than one open Querier.

That was 1:1 copied from db.go ;p  Since I don't get what it means, I will remove it.